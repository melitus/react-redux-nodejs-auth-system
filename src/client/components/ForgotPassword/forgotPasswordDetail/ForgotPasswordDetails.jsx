// @flow
import React from 'react';
import { Field, FormSection } from 'redux-form';
import type { Node } from "react";

import { Input } from '../../Field/index';

type ForgotPasswordFormDetailsProp = {
  email?: number,
};

const ForgotPasswordDetails = ({ email }: ForgotPasswordFormDetailsProp) => (
  <FormSection name="forgotPassword">
    <label target="forgotPassword.forgotPasswordDetails.email">Email</label>
    <Field
      id="forgotPassword.forgotPasswordDetails.email"
      name="email"
      component={Input}
      placeholder="Email"
      type="email"
      value={email}
    />
  </FormSection>
);

ForgotPasswordDetails.defaultProps = {
  email: '',
};

export default ForgotPasswordDetails;
