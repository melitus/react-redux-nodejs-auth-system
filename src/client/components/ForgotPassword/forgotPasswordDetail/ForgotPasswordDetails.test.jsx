import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import ForgotPasswordDetails from './ForgotPasswordDetails';

enzyme.configure({ adapter: new Adapter() });

describe('<ForgotPasswordDetails />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <ForgotPasswordDetails />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
