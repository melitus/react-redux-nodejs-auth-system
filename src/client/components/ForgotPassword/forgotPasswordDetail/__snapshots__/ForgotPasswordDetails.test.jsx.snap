exports[`<ForgotPasswordDetails /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPasswordDetails
    password="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label
          target="forgotPassword.forgotPasswordDetails.email">
          Email
</label>,
        <Field
          component={[Function]}
          id="forgotPassword.forgotPasswordDetails.email"
          name="email"
          placeholder="Email"
          type="email"
          value={undefined} />,
      ],
      "component": "div",
      "name": "forgotPassword",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email",
          "target": "forgotPassword.forgotPasswordDetails.email",
        },
        "ref": null,
        "rendered": "Email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "id": "forgotPassword.forgotPasswordDetails.email",
          "name": "email",
          "placeholder": "Email",
          "type": "email",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label
            target="forgotPassword.forgotPasswordDetails.email">
            Email
</label>,
          <Field
            component={[Function]}
            id="forgotPassword.forgotPasswordDetails.email"
            name="email"
            placeholder="Email"
            type="email"
            value={undefined} />,
        ],
        "component": "div",
        "name": "forgotPassword",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
            "target": "forgotPassword.forgotPasswordDetails.email",
          },
          "ref": null,
          "rendered": "Email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "id": "forgotPassword.forgotPasswordDetails.email",
            "name": "email",
            "placeholder": "Email",
            "type": "email",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
